From fa61f5b55ad833edf5d60aae3effe88cf2e4a9a5 Mon Sep 17 00:00:00 2001
From: Bo Huang <bo.b.huang@intel.com>
Date: Mon, 22 Apr 2013 21:45:44 +0800
Subject: [PORT FROM MAIN]Add ro.adb.secure into main.mk and enable it for 4.2.2

BZ: 100831

Android 4.2.2 introduce new feature for USB debugging.
http://android-developers.blogspot.com/2013/02/security-enhancements-in-jelly-bean.html
Enable ro.adb.secure in main.mk for user build only.

Change-Id: I9c3a28c52db6162b87c6dab087250da81e156735
Orig-Change-Id: I3b838f9a22e0fe1c8a318ec84ca7c86b7f6041d7
Category: aosp improvement
Domain: USB ADB
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Bo Huang <bo.b.huang@intel.com>
---
 core/main.mk |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/core/main.mk b/core/main.mk
index d4a5f5c..ecd8fd7 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -305,6 +305,8 @@ ifneq (,$(user_variant))
   else
     # Disable debugging in plain user builds.
     enable_target_debugging :=
+    # Enable secure USB debugging in plain user builds.
+    ADDITIONAL_DEFAULT_PROPERTIES += ro.adb.secure=1
   endif
 
   # Turn on Dalvik preoptimization for user builds, but only if not
-- 
1.7.4.1

